src/fortranscanner.l

```diff
@@ -672,11 +675,9 @@ extends{ARGS}                           {
                                         }
 public                                  {
                                           yyextra->current->protection = Public;
-                                          yyextra->typeProtection = Public;
                                         }
 private                                 {
                                           yyextra->current->protection = Private;
-                                          yyextra->typeProtection = Private;
```

Comment:
Why is the type protection removed here?

Reply:
Same as above, this only refers to the type itself, not its contained members...